lib/Demangling/CMakeLists.txt

```diff
@@ -11,3 +11,18 @@ add_swift_host_library(swiftDemangling STATIC
 target_compile_definitions(swiftDemangling PRIVATE
   LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING=1)
 
+# Runtime libraries that depend on swiftDemangling should redefine
+# SWIFT_INLINE_NAMESPACE to override the identifier that will be used for the
+# inline namespace defined by this library. This keeps the demangling symbols
+# in those libraries distinct from those in the compiler, which prevents ODR
+# violations in certain contexts; for example, on platforms that support
+# statically linking the Swift standard library, it allows this to happen
+# safely when the binary also links in compiler libraries that may not match
+# exactly with the runtime version.
+#
+# The name "cins" ("Compiler Inline NameSpace") was chosen instead of a word
+# like "compiler" in order to be less likely to collide with a real symbol
+# that someone might want to use, and because it does not appear in source
+# code.
+target_compile_definitions(swiftDemangling PUBLIC
+  SWIFT_INLINE_NAMESPACE=cins)
```

Comment:
I'm not attached to these specific names, but I guess I wanted to make sure that if someone did something reasonable down the line, like `namespace swift { namespace compiler {`, it was clear that that was distinct from this macro-hidden namespace.

I pushed an amended commit with `_compiler`, since `__compiler` would technically be a reserved word IIRC.

Reply:
Yeah, `_compiler` should be fine.  Ah, okay, I see what you mean - its merely about cognitive costs.